SomerenDAL/BaseDao.cs

```diff
@@ -17,52 +18,39 @@ protected SqlConnection OpenConnection()
             return connection;
         }
 
-        /// <summary>
-        /// For Create/Update/Delete Queries.
-        /// </summary>
-        protected void ExecuteEditQuery(string query, SqlParameter[] sqlParameters)
+        protected List<T> ReadTable<T>(DataTable dataTable, Func<DataRow, T> ReadRow)
         {
-            try
-            {
-                using (SqlConnection connection = OpenConnection())
-                {
-                    using (SqlCommand command = CreateCommand(connection, query, sqlParameters))
-                    {
-                        command.ExecuteNonQuery();
-                    }
-                }
-            }
-            catch (SqlException ex)
+            List<T> list = new List<T>();
+
+            foreach (DataRow dr in dataTable.Rows)
             {
-                HandleSqlException(ex);
+                T t = ReadRow(dr);
```

Comment:
I believe we have agreed on changing the "t" to "record" but "record" is reserved name is C# and I offer to name it "item" instead.

Reply:
Good spot. Just changed it to item as you suggested